scene/gui/tree.cpp

```diff
@@ -3561,8 +3569,13 @@ void Tree::gui_input(const Ref<InputEvent> &p_event) {
 							mpos.x -= icon_size_x;
 						}
 						if (rect.has_point(mpos)) {
-							if (!edit_selected()) {
-								emit_signal(SNAME("item_double_clicked"));
+							switch (activate_mode) {
+								case EMIT_SIGNAL:
+									emit_signal(SNAME("item_double_clicked"));
+									break;
+								case INLINE_EDIT:
+									edit_selected();
```

Comment:
Probably should fallback to emit signal if the item is uneditable, to preserve previous behavior.